### PR TITLE
Get a list of operators which exited the system (either due to opt-out, churn, or booting) within the last 14 days.

### PR DESCRIPTION
**Pull request by wmb-software-consulting**
## Why are these changes needed?

After an operator is no longer a member of EigenDA network (due to opt-out, churn or booting), they will stop receiving dispersal traffic (i.e. stop attesting the blobs). However, they are still held responsible for serving the blobs that they have signed until the blobs get expired (14 days after signing).

## Checks

- [x] I've made sure the lint is passing in this PR.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, in that case, please comment that they are not relevant.
- Testing Strategy
   - [x] Unit tests
   - [ ] Integration tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by jianoaix — COMMENTED** [review]
Could you test it locally and verify it works?

File: disperser/dataapi/subgraph/api.go [hunk 1]
```diff
@@ -2,23 +2,28 @@ package subgraph
...
+	maxEntriesPerQuery = 1000
 )
 
 type (
 	Api interface {
 		QueryBatches(ctx context.Context, descending bool, orderByField string, first, skip int) ([]*Batches, error)
-		QueryOperators(ctx context.Context, first int) ([]*OperatorRegistered, error)
+		QueryOperators(ctx context.Context, first int) ([]*Operator, error)
 		QueryBatchNonSigningOperatorIdsInInterval(ctx context.Context, intervalSeconds int64) ([]*BatchNonSigningOperatorIds, error)
+		QueryDeregisteredOperatorsGreaterThanBlockTimestampWithPagination(ctx context.Context, blockTimestamp uint64) ([]*Operator, error)
```
[the remark this answers is not in the document]
Comment (wmb-software-consulting) [reply disperser/dataapi/subgraph/api.go:25]: Trying to follow code base pattern, any suggestion?

File: disperser/dataapi/subgraph/api.go [hunk 2]
```diff
@@ -97,9 +102,49 @@ func (a *api) QueryBatchNonSigningOperatorIdsInInterval(ctx context.Context, int
...
+	for {
+		variables["first"] = graphql.Int(maxEntriesPerQuery)
+		variables["skip"] = graphql.Int(skip)
+
+		err := a.operatorStateGql.Query(ctx, &result, variables)
+		if err != nil {
+			return nil, err
+		}
+
+		if len(result.OperatorDeregistereds) == 0 {
```
[the remark this answers is not in the document]
Comment (wmb-software-consulting) [reply disperser/dataapi/subgraph/api.go:128]: Ok I will fetch all of them in one go

**Comment by wmb-software-consulting** [comment]
> Could you test it locally and verify it works?

Tested with  [https://subgraph.satsuma-prod.com](https://subgraph.satsuma-prod.com/eigenlabs/eigenda-operator-state-goerli/playground)](https://subgraph.satsuma-prod.com/027e731a6242/eigenlabs/eigenda-operator-state-goerli/api)

File: disperser/dataapi/subgraph/api.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (jianoaix) [reply disperser/dataapi/subgraph/api.go:128]: If there is benefit, you can keep it fetching in multiple round.

**Review by jianoaix — APPROVED** [review]

**Pull request merged**